dpnp/dpnp_iface_arraycreation.py

```diff
@@ -1435,14 +1456,30 @@ def meshgrid(*xi, copy=True, sparse=False, indexing="xy"):
 
     if not all((isinstance(x, (dpnp.ndarray, dpt.usm_ndarray)) for x in xi)):
         pass
-    elif indexing not in ["ij", "xy"]:
-        pass
-    elif copy is not True:
-        pass
-    elif sparse is not False:
-        pass
     else:
-        return dpnp_container.meshgrid(*xi, indexing=indexing)
+        ndim = len(xi)
+
+        if indexing not in ["xy", "ij"]:
+            raise ValueError(
+                "Unrecognized indexing keyword value, expecting 'xy' or 'ij'."
+            )
+
+        s0 = (1,) * ndim
+        output = [
+            x.reshape(s0[:i] + (-1,) + s0[i + 1 :]) for i, x in enumerate(xi)
+        ]
+
+        if indexing == "xy" and ndim > 1:
+            output[0].reshape((1, -1) + s0[2:])
+            output[1].reshape((-1, 1) + s0[2:])
+
+        if not sparse:
+            output = dpnp.broadcast_arrays(*output)
+
+        if copy:
+            output = tuple(x.copy() for x in output)
```

Comment:
as I can see numpy and cupy return a list here, not a tuple
```suggestion
            output = [x.copy() for x in output]
```